docs/content/docs/concepts/signals.md

```diff
@@ -6,6 +6,92 @@ discussion: 5469
+
+For more information on using `ContainerSchema` to create objects please see [Data modeling](https://fluidframework.com/docs/build/data-modeling/).
+
+## API
+`Signaler` provides a few simple methods to send signals and add/remove listeners to specific signals as well:
+- `submitSignal(signalName: string, payload?: Jsonable)` - Sends a signal with a payload to its connected listeners
+- `onSignal(signalName: string, listener: SignalListener)` - Adds a listener for the specified signal. Similar behavior as EventEmitter's `on` method.
+- `offSignal(signalName: string, listener: SignalListener)` - Removes a listener for the specified signal. Similar behavior as EventEmitter's `off` method.
+
+## Common Patterns
+### Signal Request
+When a client joins a collaboration session, they may need to receive information about the current state immediately after connecting the container.  To support this, they can request a specific signal be sent to them from other connected clients. For example, in the [PresenceTracker](https://github.com/microsoft/FluidFramework/tree/main/examples/data-objects/presence-tracker) example we define a "focusRequest" signal type that a newly joining client uses to request the focus-state of each currently connected client:
+
+```typescript
+private static readonly focusRequestType = "focusRequest";
+```
+
+```typescript
+container.on("connected", () => {
+    this.signaler.submitSignal(FocusTracker.focusRequestType);
+});
+```
+
+The connected clients are listening to this focus request signal, and they respond with their current focus state:
+
+```typescript
+this.signaler.onSignal(FocusTracker.focusRequestType, () => {
+    this.sendFocusSignal(document.hasFocus());
+});
+```
+This pattern adds cost however, as it forces every connected client to generate a signal.  Consider whether your scenario can be satisfied by receiving the signals naturally over time instead of requesting the information up-front. The mouse tracking in [PresenceTracker](https://github.com/microsoft/FluidFramework/tree/main/examples/data-objects/presence-tracker) is an example where a newly connecting client does not request current state. Since mouse movements are frequent, the newly connecting client can instead simply wait to receive other users' mouse positions on their next mousemove event.
+### Grouping Signal Types
+
+Rather than submitting multiple signals in response to an event, it is more cost-effective to submit one combined signal for that event and listen to that single signal instead. For example, imagine an application using the `Signal Request` pattern where a newly connected client requests for the color, focus state, and currently selected object of every other connected client on the page. If you submit a signal for each type of data requested, it would look something like this:
+
+```typescript
+container.on("connected", () => {
+    this.signaler.submitSignal("colorRequest");
```

Comment:
Need to explain briefly where these signal types, like "colorRequest" are defined.

Reply:
They are defined at the same time they are submitted. "colorRequest" is the signal name being sent to indicate the client is requesting the color of other users.

Reply:
@WillieHabimana I'm still puzzled. The "colorRequest" signal is sent to something. So that something must have a definition of what "colorRequest" means. Where is that definition?

Reply:
Hmmm well It's being sent to other listening clients that will respond with submitting a signal with their color. Would having something like this help:
```typescript
this.signaler.onSignal("colorRequest", (clientId, local, payload) => {
            const clientIdColorMap = this.colorMap.get(this.audience.getMyself().userId);
            const clientColor =  clientIdColorMap.get(clientId);
            this.submitSignal("colorSignalType", {color: clientColor});
        });
```
I think the code snippet was just to give an idea of the best practice of what to do when sending multiple signals at once. Do the code snippets make the concept more confusing? If so, I think they could be removed.

Reply:
Yes. This additional snippet is just what I had in mind.

Reply:
Perfect, sounds good!